lib/Sema/CSFix.h

```diff
@@ -291,7 +291,10 @@ class ConstraintFix {
   virtual bool diagnose(const Solution &solution,
                         bool asNote = false) const = 0;
 
-  virtual bool diagnoseForAmbiguity(ArrayRef<Solution> solutions) const { return false; }
+  virtual bool diagnoseForAmbiguity(ArrayRef<Solution> solutions,
+                                    ArrayRef<ConstraintFix *> fixes) const {
```

Comment:
Could you please adjust this in way that it takes `ArrayRef<std::pair<const Solution &, Constraint Fix *>>`. That would future proof this interface a bit...